pkg/types/querybuildertypes/querybuildertypesv5/trace_operator.go

```diff
@@ -0,0 +1,441 @@
+package querybuildertypesv5
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/SigNoz/signoz/pkg/errors"
+	"github.com/SigNoz/signoz/pkg/types/telemetrytypes"
+	"github.com/SigNoz/signoz/pkg/valuer"
+)
+
+// TraceOperatorType defines the type of trace operator
+type TraceOperatorType struct{ valuer.String }
+
+var (
+	TraceOperatorDirectDescendant   = TraceOperatorType{valuer.NewString("=>")}
+	TraceOperatorIndirectDescendant = TraceOperatorType{valuer.NewString("->")}
+	TraceOperatorAnd                = TraceOperatorType{valuer.NewString("&&")}
+	TraceOperatorOr                 = TraceOperatorType{valuer.NewString("||")}
+	TraceOperatorNot                = TraceOperatorType{valuer.NewString("NOT")}
+	TraceOperatorExclude            = TraceOperatorType{valuer.NewString("NOT")}
```

Comment:
These are redundant

Reply:
A `NOT` is negation and is used differently depending on context. `A NOT B` is `A AND (NOT B)`. Please explain how they are different.

Reply:
That doesn't answer the question. I am not able to follow how are they different operators?

Reply:
There is no such thing as binary NOT. A negation is always a unary https://en.wikipedia.org/wiki/Negation. I will end this discussion here to keep things moving, especially since this is not end-user facing.